### DISCUSSION TITLE
Installation

### DISCUSSION THREAD

**Discussion by giscus** [hbstack/site#226] [General]
# Installation

We provide a starter theme template for you to create a HB site quickly.

https://hbstack.dev/en/docs/getting-started/installation/

**Comment by TheFriendlyCoder** [comment]
Do you have plans to add support for installing and upgrading the theme using a Git submodule or Hugo module, similar to the way [your previous theme](https://hbs.razonyang.com/v1/en/docs/getting-started/installation/hugo-module/) worked? I am relatively new to Hugo and your theme, and it was so much easier to get your v1 HBS theme set up and running.

**Reply by razonyang** [reply]
Actually I can't, this theme is designed to be modular (only work with Hugo modules), I don't know how it works with Git submodule, the hard part is the dependency's resolver of Git submodule, since the theme requires a lot of modules.

**Reply by TheFriendlyCoder** [reply]
So does that mean you can install this theme using Hugo modules then? Do you have any docs on how that's done?

**Reply by razonyang** [reply]
The current [page](https://hbstack.dev/en/docs/getting-started/installation/) is the docs you're looking for, you may need to install the [build tools](https://hbstack.dev/en/docs/getting-started/prerequisites/#build-tools) before executing the instructions.

**Reply by TheFriendlyCoder** [reply]
sorry if I'm missing something obvious, but the installation page you referenced talks about cloning an existing repository as the starting point for setting up the theme, and doesn't mention anywhere about using `hugo mod get` commands. Is that perhaps implied somewhere in how the installation guide is worded and I am just missing it?

Again, sorry if this is a silly / obvious question. I'm relatively new to Hugo and am still trying to learn how things work.

**Reply by razonyang** [reply]
Never mind.

> cloning an existing repository as the starting point for setting up the theme.

Yep, the repo is created for setting up the theme as simple as possible.

> doesn't mention anywhere about using hugo mod get commands

Since the template repo already done some setup for you, such as the Hugo modules, and the [`npm run dev` or `npm run prod`](https://github.com/hbstack/theme/blob/31d8401e4c53653177f819a6f37ebb8e09fb460e/package.json#L5-L6) will start a Hugo server that download modules (`hugo mod get`), build and preview the site, you don't have to execute `hugo mod get` manually, unless [upgrading modules](https://hbstack.dev/en/modules/overview/#upgrade-modules).

**Reply by TheFriendlyCoder** [reply]
I see. Thanks for the explanation.

So, do you have any docs that explain how to add the template to an existing Hugo site? Cloning another repo and then trying to merge it together with another existing one definitely seems more risky / complicated than just adding a Hugo module.

Suggestion: it may be helpful for newcomers if you were to make a note somewhere on your installation page mentioning something about how the sample repository already has the Hugo modules set up for you, and maybe include a link to the module upgrading page in there as well for reference.

**Reply by razonyang** [reply]
Added it, see https://hbstack.dev/en/docs/getting-started/install-on-existing-sites/.

**Comment by go-alex** [comment]
hugo 0.120.0、0.120.1 will cause problems with the homepage style. The class of the main module is wrong.
The highest version currently available is 0.119

**Reply by razonyang** [reply]
Thanks for your feedback, I created patches for fixing this, https://hbstack.dev/en/2023/10/broken-styles-when-building-site-via-hugo-v0.120.0-or-later/.

**Comment by diogoguilhen** [comment]
hello there due installation i found 1 alert by npm ...
 \blog>npm audit
# npm audit report

yaml  2.0.0-5 - 2.2.1
Severity: high
Uncaught Exception in yaml - https://github.com/advisories/GHSA-f9xv-q969-pqx4
fix available via `npm audit fix`
node_modules/yaml

1 high severity vulnerability

To address all issues, run:
  npm audit fix

**Reply by razonyang** [reply]
Thanks for your feedback, fixed it via https://github.com/hbstack/theme/commit/3ed824778012190115f40386ff28489ef34a6289, you can follow the instruction (`npm audit fix`) to fix it on your existing sites.
